control/controlhttp/client.go

```diff
@@ -272,6 +272,18 @@ func (d *Dialer) forceNoise443() bool {
 	if forceNoise443() {
 		return true
 	}
+
+	if d.HealthTracker.LastNoiseDialWasRecent() {
```

Comment:
I don't love this because e.g. netmon changes that result in us re-dialing control will trip this heuristic, but it's "safe" in that it'll just be slightly more expensive, not wrong. Still want to do a healthcheck eventually, but this seems safe for now.